node/src/components/deploy_buffer.rs

```diff
@@ -92,14 +113,25 @@ pub(crate) struct DeployBuffer {
 
 impl DeployBuffer {
     /// Creates a new, empty deploy buffer instance.
-    pub(crate) fn new(block_max_deploy_count: usize) -> Self {
-        DeployBuffer {
+    pub(crate) fn new<REv>(
+        event_queue: EventQueueHandle<REv>,
```

Comment:
Why not pass in an `EffectBuilder` instead of a queue handle?

Reply:
Sure, makes sense, I think I was using `SmallNetwork` as an example originally, but since there's no good reason to hold onto the queue handle I can do that.